pdsfile/__init__.py

```diff
@@ -189,25 +168,21 @@ def use_shelves_only(status=True):
     presence in the infoshelf files first. Search the file system only if a
     shelf is missing."""
 
-    global SHELVES_ONLY
-
-    SHELVES_ONLY = status
+    cfg.SHELVES_ONLY = status
 
 ################################################################################
 # How to handle missing shelf files
 ################################################################################
 
-SHELVES_REQUIRED = False
+# SHELVES_REQUIRED = False
 
 def require_shelves(status=True):
     """Call before preload(). Status=True to raise exceptions when shelf files
     are missing or incomplete. Otherwise, missing shelf info is only logged as a
     warning instead.
     """
 
-    global SHELVES_REQUIRED
-
-    SHELVES_REQUIRED = status
+    cfg.SHELVES_REQUIRED = status
 
 ################################################################################
 # Memcached and other cache support
```

Comment:
I think it would be really nice to finally separate out all of this cache and preload management from the rest of pdsfile, and just have this file contain the PdsFile class definition.

Reply:
Fixed.